packages/documentation/docs/SAG/SAG-for-starknet.md

```diff
@@ -1,6 +1,135 @@
 ---
 id: SAG-starknet
 title: "SAG for Starknet"
---- 
+---
+
+# SAG for Starknet
+
+The **SAG for Starknet** implementation enables verification of Spontaneous Anonymous Group (SAG) signatures on the Starknet platform. This implementation leverages Starknet-specific optimizations to perform ring signature verification efficiently on-chain, supporting both SECP256K1 and ED25519 elliptic curves.
+
+The SAG verification functionality on Starknet is provided by the `starknet-sag-ts` package from [Alice's Ring](https://github.com/Cypher-Laboratory/Alice-s-Ring-SAG-TS), a TypeScript library for creating SAG signatures.
+
+## Key Differences in the Starknet Implementation
+
+The Starknet version of the SAG verifier shares the same fundamental logic as the EVM version, with a few important adjustments to optimize for the Starknet ecosystem:
+
+1. **Poseidon Hash Function**: The signer and verifier use the Poseidon hashing function, which is computationally cheaper on Starknet compared to traditional hashing algorithms.
+
+2. **Garaga for Multi-Scalar Multiplication (MSM)**: The implementation is using Garaga allowing efficient on-chain multi-scalar multiplication (MSM). The process incorporates hints passed alongside the ring signature to perform MSM efficiently.
+
+## Generating a Ring Signature for Starknet
+
+Here's how to generate a ring signature for starknet using Alice's Ring in TypeScript:
+
+```typescript
+import {
+  RingSignature,
+  Curve,
+  CurveName,
+  Point,
+} from "@cypher-laboratory/alicesring-sag-starknet";
+
+const curve = new Curve(CurveName.SECP256K1);
+const ring: Point[] = []; // Your ring of public keys
```

Comment:
might be good to add 2 or 3 public keys so the user can see how they should be encoded (compressed to 33Bytes I guess)

Reply:
done